### ISSUE TITLE
Vacuuming tables that don't have sort keys

### ISSUE THREAD

**Issue by MBloom** [awslabs/amazon-redshift-utils#182]
I have recently come across an issue when I run `ANALYZE` on a table that does not have a sort key defined. I have noticed that the `stats_off` percent will remain unchanged after the query has executed. This is likely due to the table containing old records that have been deleted. 

Ideally, the analyze-vacuum-schema.py script would first cleanup this table by running `VACUUM`. 
There is currently a [clause in the vacuum query](https://github.com/awslabs/amazon-redshift-utils/blob/master/src/AnalyzeVacuumUtility/analyze-vacuum-schema.py#L349) to check for a null sort key. However, since `unsorted` is null whenever `sortkey1` is null, the inner clause will never be True for tables that do not have a sort key defined. Thus, the table will never be vacuumed.

Fixing the query to be something like this should work:
```sql
...

WHERE "schema" = '{schema}'
    AND (
            (
                (
                --If the size of the table is less than the max_table_size_mb then
                --run vacuum based on condition:
                -- >min_unsorted_pct AND >deleted_pct
                    ((size < {max_table_size}) AND (unsorted > {min_unsorted_pct}
                        OR empty > {deleted_pct}))
                    OR
                --If the size of the table is greater than the max_table_size_mb then
                --run vacuum based on condition:
                -- >min_unsorted_pct AND < max_unsorted_pct AND >deleted_pct
                --This is to avoid big table with large unsorted_pct
                     ((size > {max_table_size}) AND (unsorted > {min_unsorted_pct}
                        AND unsorted < {deleted_pct} ))
                 )
                AND sortkey1 not ilike 'INTERLEAVED%%'
            )
            OR sortkey1 IS NULL
        )
```

**Comment by timjell** [comment]
In the current releases, tables are defined with a sortkey, otherwise it is an auto sortkey. This shows as sortkey1 = 'AUTO(SORTKEY)' value. To fix this issue, we need to wrap the 'unsorted' field into an NVL, since this is always null even for an auto sortkey. This will then pick up tables without a defined sortkey, but where the 'stats_off' value changes when new rows are inserted.

**Comment by timjell** [comment]
Fix to be deployed in #629. Closing this issue.
